webapp/src/services/near-nep21-util.js

```diff
@@ -0,0 +1,325 @@
+import { Contract} from 'near-api-js'
+import getConfig from '../config'
+
+const nearConfig = getConfig(process.env.NODE_ENV || 'development')
+const attachNear1 = '300000000000000';
+const attachNear2 = '60000000000000000000000';
+
+export async function getBalanceNEP( contractName ) {
+
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    contractName ,
+    {
+      // View methods are read only
+      viewMethods: ['get_balance'],
+      // Change methods modify state but don't receive updated data
+      changeMethods: []
+    }
+  )
+
+  return await window.nep21.get_balance({ owner_id: window.walletConnection.getAccountId() });
+
+}
+
+export async function incAllowance( token ) {
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    token.address ,
+    {
+      // View methods are read only
+      viewMethods: [],
+      // Change methods modify state but don't receive updated data
+      changeMethods: ['inc_allowance']
+    }
+  )
+  
+  try {
+    await window.nep21.inc_allowance({ 
+      escrow_account_id: nearConfig.contractName, 
+      amount: token.amount},
+      attachNear1,
+      attachNear2
+      );
+     console.log("DONE"); 
+    return true;
+  } catch(error) {
+    return false;
+  }
+
+}
+
+export async function getAllowance( token ) {
+  const accountId = window.accountId;
+  console.log(accountId);
+  window.nep21 = await new Contract(
+    window.walletConnection.account(),
+    token.address ,
+    {
+      // View methods are read only
+      viewMethods: ['get_allowance'],
+      // Change methods modify state but don't receive updated data
+      changeMethods: []
+    }
+  )
+  const allowance = await window.nep21.get_allowance({
+    owner_id: accountId, 
+    escrow_account_id: nearConfig.contractName });
+  console.log('Allowance: ', allowance);
+  return allowance;  
+}
+
+export async function gasCheck() {
+  // Set default
+  const near_limit = 0.6;
+  const bal = (await window.walletConnection.account().getAccountBalance()).available / 1000000000000000000000000;
```

Comment:
the long number should be a constant. Please declare it a as `const NDENOM = 1e24` (we are using the same const in the backend -> near denomination). BTW - you can also use big num lib (`BN`) if precesion is important.